context/app/static/js/components/RecursiveList.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import DataPanel from './DataPanel';
+import RecursiveListLeaf from './RecursiveListLeaf';
+import { isEmptyArrayOrObject } from '../helpers/functions';
+
+const RecursivePropertyContainer = styled.div`
+  margin-top: 5px;
+  font-size: 16px;
+`;
+
+const PropertyName = styled.span`
+  font-weight: bolder;
+  font-size: 1rem;
+`;
+
+function isLeaf(property) {
+  return (typeof property === 'number' || typeof property === 'string' || typeof property === 'boolean' || isEmptyArrayOrObject(property));
+}
+
+function RecursiveList(props) {
+  const {
+    property, propertyName, isRoot, isRootChild,
+  } = props;
+
+
+  return (
+    /* eslint-disable no-nested-ternary */
+    <RecursivePropertyContainer>
+      { property ? (
+        isLeaf(property)
+          ? (
+            <RecursiveListLeaf
+              property={property}
+              propertyName={propertyName}
+              isRootChild={isRootChild}
+            />
+          ) : (
+            isRootChild
+              ? (
+                <DataPanel propertyName={propertyName} isRootChild={isRootChild}>
+                  {Object.values(property).map((objProperty, index) => (
+                    <RecursiveList
+                    // eslint-disable-next-line react/no-array-index-key
+                      key={index}
```

Comment:
Would this be any better?
```suggestion
                      key={objProperty}
```

Reply:
The key would then be one of an object, array, number or string. Could change the key to the 'propertyName' ?